### ISSUE TITLE
Backbone

### ISSUE THREAD

**Issue by MarcSchotman** [DeepSceneSeg/AdapNet-pp#1]
Hi,

I've been playing around with your AdapNet++ and SSMA and it looks really good, love the SSMA paper very well written!
Though, regarding the code, there were some troubles getting it all started as there are some things which are not that clear from the notes. So for anyone who is running in to troubles:

- For raining a AdapNet++ network I deleted this part of the code so that it does not try to initialize with anything:
```
 ckpt = tf.train.get_checkpoint_state(os.path.dirname(os.path.join(config['checkpoint'],
                                                                      'checkpoint')))
    if ckpt and ckpt.model_checkpoint_path:
        saver = tf.train.Saver(max_to_keep=1000)
        saver.restore(sess, ckpt.model_checkpoint_path)
        step = int(ckpt.model_checkpoint_path.split('/')[-1].split('-')[-1])+1
        sess.run(tf.assign(global_step, step))
        print 'Model Loaded'

    else:
        reader = tf.train.NewCheckpointReader(config['intialize'])
        var_str = reader.debug_string()
        name_var = re.findall('[A-Za-z0-9/:_]+ ', var_str)
        import_variables = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES)
        initialize_variables = {} 
        for var in import_variables: 
            if var.name+' ' in  name_var:
                initialize_variables[var.name] = var

        saver = tf.train.Saver(initialize_variables)
        saver.restore(save_path=config['intialize'], sess=sess)
        saver = tf.train.Saver(max_to_keep=1000)
        print 'Intialized'
```
and added `saver = tf.train.Saver()`. It looks to me as if there is no option for simple training. Both these methods try to initialize with some sort of pre-trained weights..
- There was no `__init__.py` file in the models folder of the SSMA repo. This results in an import error. This gave me a big headache as I didn't know about the existence or need of this file ;).
- The `convert_to_tfrecords.py` scripts are different for SSMA and AdapNet repo, a simple `len(name)` check could make them compatible
- Altered line 29 in `convert_to_tfrecords.py` to `all_list = file_handler.read().splitlines()` to get rid of the previously present `\n` in the list.
- Your images have to be cropped before you convert them.
- Your labels need to be in grayscale. (maybe implement a check?)
- The fact that you have to download this repo, make 2 models and then use these for the SSMA network is a bit cumbersome.

Now my dataset is rather small so I would like to initialize with pre-trained weights as you mention in the notes. I had troubles finding these online. Do you have a download link?

I am also having trouble getting tensorflow to use my gpu. Another environment where I have tensorflow 1.12 works fine. But tensorflow1.4 does not see my gpu for some reason... I have CUDA 9.2 and also installed this wheel: https://github.com/mind/wheels/releases/tag/tf1.4-gpu-cuda9 with no result. You have any tips for this?

Cheers

**Comment by avalada** [comment]
Thank you for the interest in our work.

- We have updated the train.py script to support training from scratch. You can train from scratch by not mentioning the initialize option in the .config file.
- The model folder of the SSMA repo does have the __init__.py file.
- Separate scripts for creating the tfrecords was intentional as people might want to only train the unimodal AdapNet++ or AdapNet so they dont have to create the tfrecords with multimodal data. We have added a note on this to the README.
- We have fixed \n issue.
- All the augmentations whether it's cropping, scaling or flipping needs to be done before converting to tfrecords. No augmentation takes place during training. 
- We have added channel check for label. The labels used for training are generally one channel. 
- You dont necessarily have to train the unimodal models first and then train the SSMA with pretrained weights from the unimodal model. You can also directly train the SSMA without any pretrained model initialization, but this gives a lower performance :)  The SSMA repo does not depend on the Adapnet++ repo. If you want to train the SSMA from scratch, you can remove the chekpoint1 and checkpoint2 option in the .config file and you are set to go.

You can find pretrained weights in the init_checkpoint folder. You should probably augment your data beforehand to increase the number of training images.

Maybe try building tensorflow from source.

**Comment by MarcSchotman** [comment]
Thanks for the quick response !

Regarding the tensorflow issue:  When rewriting the print statements your code seems to work fine with tensorflow 1.12 and python3 :).
